File: samples/sandbox/app/src/main/java/com/scichart/docsandbox/examples/java/chartModifier2D/ZoomAndPanXAxisDragModifier.java
package com.scichart.docsandbox.examples.java.chartModifier2D;

import androidx.annotation.NonNull;

import com.scichart.charting.ClipMode;
import com.scichart.charting.ClipModeTarget;
import com.scichart.charting.modifiers.AxisDragModifierBase;
import com.scichart.charting.modifiers.XAxisDragModifier;
import com.scichart.charting.visuals.SciChartSurface;
import com.scichart.docsandbox.core.ExampleDefinition;
import com.scichart.docsandbox.examples.base.SingleChart2DFragment;

@ExampleDefinition()
public class ZoomAndPanXAxisDragModifier extends SingleChart2DFragment {
    @Override
    protected void initExample(@NonNull SciChartSurface surface) {}

    void addXAxisDragModifier(@NonNull SciChartSurface surface) {
        // <AddXAxisDragModifier>
        // Assume a surface has been created and configured somewhere
        // Create a Modifier
        final XAxisDragModifier xAxisDragModifier = new XAxisDragModifier();
        xAxisDragModifier.setDragMode(AxisDragModifierBase.AxisDragMode.Pan);
        xAxisDragModifier.setClipModeX(ClipMode.StretchAtExtents);
        xAxisDragModifier.setClipModeTargetX(ClipModeTarget.MaximumRange);

        // Add the modifier to the surface
        surface.getChartModifiers().add(xAxisDragModifier);
        // </AddXAxisDragModifier>
    }
}

File: samples/sandbox/app/src/main/java/com/scichart/docsandbox/examples/kotlin/chartModifier2D/ZoomAndPanXAxisDragModifier.kt
package com.scichart.docsandbox.examples.kotlin.chartModifier2D

import com.scichart.charting.ClipMode
import com.scichart.charting.ClipModeTarget
import com.scichart.charting.modifiers.AxisDragModifierBase
import com.scichart.charting.modifiers.XAxisDragModifier
import com.scichart.charting.visuals.SciChartSurface
import com.scichart.docsandbox.core.ExampleDefinition
import com.scichart.docsandbox.examples.base.SingleChart2DFragment

@ExampleDefinition()
class ZoomAndPanXAxisDragModifier : SingleChart2DFragment() {
    override fun initExample(surface: SciChartSurface) {}

    fun addXAxisDragModifier(surface: SciChartSurface) {
        // <AddXAxisDragModifier>
        // Assume a surface has been created and configured somewhere
        // Create a Modifier
        val xAxisDragModifier = XAxisDragModifier()
        xAxisDragModifier.dragMode = AxisDragModifierBase.AxisDragMode.Pan
        xAxisDragModifier.clipModeX = ClipMode.StretchAtExtents
        xAxisDragModifier.clipModeTargetX = ClipModeTarget.MaximumRange

        // Add the modifier to the surface
        surface.chartModifiers.add(xAxisDragModifier)
        // </AddXAxisDragModifier>
    }
}

